mm2src/lp_ordermatch.rs

```diff
@@ -3703,34 +3901,168 @@ fn my_taker_order_file_path(ctx: &MmArc, uuid: &Uuid) -> PathBuf {
     my_taker_orders_dir(ctx).join(format!("{}.json", uuid))
 }
 
-fn save_my_maker_order(ctx: &MmArc, order: &MakerOrder) {
+fn my_order_history_file_path(ctx: &MmArc, uuid: &Uuid) -> PathBuf {
+    my_orders_history_dir(ctx).join(format!("{}.json", uuid))
+}
+
+// TODO: history field can be part of the MakerOrder Struct Directly as an option,
+// But have to be sure it doesn't break any backward compatability
+// + it will make any response with updated maker order very long
+#[derive(Clone, Deserialize, Serialize)]
+struct UpdateMakerOrderHistory {
+    order: MakerOrder,
+    history: Vec<MakerOrder>,
```

Comment:
I prefer having the optional `changes_history` field instead of this separate struct. It leads to
```rust
if let Ok(mut original_order) = json::from_slice::<UpdateMakerOrderHistory>(&slurp(&path)) {
            original_order.history.insert(0, original_order.order);
            original_order.order = order.clone();
            let content = json::to_vec(&original_order).unwrap();
            write(&path, &content).unwrap();
        } else if let Ok(original_order) = json::from_slice::<MakerOrder>(&slurp(&path)) {
```
pattern that doesn't look good to me.

Also, `save_my_new_maker_order` and `save_maker_order_on_update` will allow avoiding `status == MakerOrderStatus::Updated` check and simply use something like 
```rust
order.get_or_insert_default.push(new_change)
```

Also, saving the entire old order in history might be too much. It might be a separate struct or even `MakerOrderUpdateReq` skipping serialization of None fields.